Scripts/Utilities/Credentials.cs

```diff
@@ -124,13 +124,14 @@ public Credentials(string username, string password, string url = null)
         }
 
         /// <summary>
-        /// Constructor that takes an authentication token created by the user or an ApiKey. 
+        /// Constructor that takes an authentication token created by the user or an ApiKey.
+        /// If no URL is set then default to the non-IAM Visual Recognition endpoint.
         /// </summary>
         /// <param name="url">The service endpoint.</param>
         public Credentials(string apiKey, string url = null)
         {
             ApiKey = apiKey;
-            Url = url;
+            Url = !string.IsNullOrEmpty(url) ? url : "https://gateway-a.watsonplatform.net/visual-recognition/api";
```

Comment:
Maybe it's obvious, but why are we setting to `https://gateway-a.watsonplatform.net/visual-recognition/api`? How are we making sure that iam user will not end here?

Reply:
We set the gateway-a endpoint here If they authenticate with the old apikey. I guess we can try to stop IAM users from putting their iam apikey here. What do you think? Maybe check the length of the apikey and error if it is not the correct length? We have documentation to authenticate using IAM as well but I am open to suggestions?

Reply:
Aren't we getting the `url` from VCAP services even for IAM? Because credentials have `url` usually. Even if it's CF or RC.

<img width="787" alt="screen shot 2018-05-31 at 12 05 56 pm" src="https://user-images.githubusercontent.com/16810846/40793661-03332e68-64cb-11e8-95ad-98c55af91cbb.png">

I don't remember what was the length for old cf `api_key` :( Maybe we can do that.

Reply:
PR looks good otherwise

Reply:
Yes we are, but we don't always use the url and use the default urls. We want to make sure that if they authenticate with apikey that the url goes to `gateway-a`

Reply:
but yes if you use the url provided in the credentials it should work either way

Reply:
great. i am approving